aten/src/ATen/cpu/vec256/vec256_qint.h

```diff
@@ -154,6 +154,223 @@ inline void __attribute__((always_inline)) QuantizeAvx2(
 #endif
 }
 
+#if defined(__AVX2__)
+
+// utility functions for horizontal sums and sums of squares
+
+// horizontal sums signed i64, overflow unsafe
+// x = (y3, y2, y1, y0)
+int64_t custom_mm256_hsum_epi64_ignore_overflow(__m256i x) {
```

Comment:
p.s. I tried to put these in a separate header but got a bunch of linking errors with "multiple functions of the same name" defined across various build flags.  Let me know if there is something special that needs to be done for a new header in this dir.